exla/c_src/exla/exla_client.cc

```diff
@@ -136,14 +138,8 @@ xla::StatusOr<ERL_NIF_TERM> ExlaExecutable::Run(ErlNifEnv* env,
   terms.reserve(input_buffers.size());
 
   for (auto buf : input_buffers) {
+    EXLA_EFFECT_OR_RETURN_NIF(buf->BlockHostUntilReady(), env);
     pjrt_buffers.push_back(buf->buffer());
-    // If the buffer was not received as a resource (e.g. we converted
-    // it from a binary to a buffer), we need to make it a resource so
-    // it's tracked and GC'ed along with other buffers that are no longer
-    // in use
-    if (buf->release_after_run()) {
-      terms.push_back(nif::make<ExlaBuffer*>(env, buf));
-    }
```

Comment:
@seanmor5 please check if we indeed can remove this. I think the goal of this code was to make sure the binaries are not relocated but, as we mentioned, there is no such guarantee.

Reply:
I don't think we can remove this, the purpose of this is to avoid memory leaks. If we move a binary to the GPU, then it needs to be cleaned up immediately after execution OR we take this approach which allows the GC to do it for us.